### PR TITLE
Adds Core Navigation Analytics

### PR DESCRIPTION
**Pull request by vinnyjth**
## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

Screen Shot 2020-02-14 at 10 51 08 AM

### CODE REVIEW

**Comment by vinnyjth** [comment]
@conrad-vanl Tagging you here to show you what I came up with. Still need to double check to make sure I didn't break any tests and make sure we are scrubbing IPs

File: packages/apolloschurchapp/src/index.js [hunk 1]
```diff
@@ -65,11 +66,16 @@ const App = () => (
   <Providers>
     <BackgroundView>
       <AppStatusBar barStyle="dark-content" />
-      <AppContainer
-        ref={(navigatorRef) => {
-          NavigationService.setTopLevelNavigator(navigatorRef);
-        }}
-      />
+      <CoreNavigationAnalytics>
+        {(props) => (
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/src/index.js]: Interesting!

**Comment by conrad-vanl** [comment]
@vinnyjth my goal on this is to hold off on merging until we figure out this interactions stuff a little bit.

**Comment by conrad-vanl** [comment]
@vinnyjth one of the things that came out of the board meeting is they'd actually like to prioritize having this merged. Could you do a pass on this and make sure it's still up to snuff with any changes we've made since? And then we can work on merging it and including it in a release sometime this cycle.

**Comment by vinnyjth** [comment]
Sure thing! I essentially need to swap some of the bits into the template repo. Is the part that we are interested in the "Navigation Analytics" or the "Core / Astronomer Analytics" or both?

**Comment by conrad-vanl** [comment]
The Core / Astronomer analytics piece mainly. Would be nice to have navigation analyitcs in that but main thing is getting some analytics feeding into astronomer from all the apps

**Comment by vinnyjth** [comment]
@conrad-vanl This PR is now caught up to our current core structure. I'll open up a PR in templates later today.

**Comment by vinnyjth** [comment]
Companion PR: https://github.com/ApollosProject/apollos-templates/pull/80

**Comment by vinnyjth** [comment]
Yep! Credentials are now in the Apollos vault.

**Pull request merged**